python/deltalake/writer.py

```diff
@@ -132,8 +132,28 @@ def write_deltalake(
     :param overwrite_schema: If True, allows updating the schema of the table.
     :param storage_options: options passed to the native delta filesystem. Unused if 'filesystem' is defined.
     """
+
     if _has_pandas and isinstance(data, pd.DataFrame):
-        data = pa.Table.from_pandas(data)
+        if schema is not None:
+            data = pa.Table.from_panda(data, schema=schema)
```

Comment:
Is this code path not tested? We might want to add a test for that.

```suggestion
            data = pa.Table.from_pandas(data, schema=schema)
```

Reply:
Thanks for the catch.  Added a test for this as well.